### PR TITLE
done with 3-GET /api/categories

### PR DESCRIPTION
**Pull request by nasben64**

### CODE REVIEW

**Review by Harrison365 — COMMENTED** [review]
Hi Nesser :) 
Please could you...
-Use a forEach to test you're array contains objects with slug and description, rather then testing index 0. 
-Remove app.use(express.json()); from app.js until you reach a POST/PATCH task
The rest looks good :) 
Please fix these things, push and send us another nchelp pr. Please don't merge until we have cheched the fixes. Thanks

File: controllers/categories.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const { selectAllCategories } = require("../models/categories");
+
+exports.getAllCategories = (req, res, next) => {
+  selectAllCategories()
+    .then((categories) => {
+      res.status(200).send(categories);
+    })
+    .catch((err) => {
+      console.log("controller error", err);
```
Comment (emilyb93) [inline controllers/categories.js]: remove console.logs

File: app.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  //next(err);
+});
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
```
Comment (emilyb93) [inline app.js]: Currently not formulating any custom errors so this should not be here yet

File: app.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const app = express();
+
+app.use(express.json());
+
+app.get("/api/categories", getAllCategories);
+
+app.all("/*", (req, res) => {
+  res.status(404).send({ msg: "Path not found" });
+  //next(err);
```
Comment (emilyb93) [inline app.js]: Remove this

**Review by emilyb93 — APPROVED** [review]
Remove some unnecessary code and then this is good to merge

**Review by Harrison365 — APPROVED** [review]
good to merge :)

**Pull request merged**